### PR TITLE
Inverse distance weigthed method for interpolation.

### PR DESCRIPTION
**Pull request by hlherrera**
In this case, we can interpolate vector layers (point, line,...) based on coordinate points and z-value of each point generating raster layers.

### CODE REVIEW

**Comment by tbonfort** [comment]
This doesn't seem ready to be applied yet, delaying to 7.2. More comments coming...

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,12 +14,12 @@ include(CheckFunctionExists)
 include(CheckIncludeFile)
 include(CheckCSourceCompiles)
 
-
 set (MapServer_VERSION_MAJOR 6)
 set (MapServer_VERSION_MINOR 5)
 set (MapServer_VERSION_REVISION 0)
 set (MapServer_VERSION_SUFFIX "")
 
+#set(CMAKE_BUILD_TYPE "DEBUG")
```
Comment (tbonfort) [inline CMakeLists.txt]: to be removed

File: README [hunk 2]
```diff
@@ -26,7 +26,43 @@ Join the MapServer user mailing list online at:
 
   http://www.mapserver.org/community/lists.html
 
- 
+
```
Comment (tbonfort) [inline README]: This does not belong in the README

File: kerneldensity.c [hunk 3]
```diff
@@ -28,7 +28,6 @@
 
 #include "mapserver.h"
 #include <float.h>
-#ifdef USE_GDAL
```
Comment (tbonfort) [inline kerneldensity.c]: Why is this removed?

File: maplexer.l [hunk 4]
```diff
@@ -1,753 +0,0 @@
-%{
```
Comment (tbonfort) [inline maplexer.l]: There should be no other change to maplexer.l than adding the idw keyword. Also please do not add the changes to maplexer.c to the pull request

File: mapraster.c [hunk 5]
```diff
@@ -63,110 +63,110 @@ extern parseResultObj yypresult; /* result of parsing, true/false */
 static int msGetClass_String( layerObj *layer, colorObj *color, const char *pixel_value )
 
 {
-  int i;
```
Comment (tbonfort) [inline mapraster.c]: please do not commit whitespace changes

File: kerneldensity.c (hunk 3 above)
Comment (hlherrera) [inline kerneldensity.c]: yes, I make that mistake.

**Comment by jmckenna** [comment]
@hlherrera I am wondering if you can re-look at this with current master, and let us know here if you are ready to proceed with these changes?

**Comment by rouault** [comment]
Ported to latest main in https://github.com/MapServer/MapServer/pull/6227

**Comment by jmckenna** [comment]
thanks @rouault and @hlherrera !!
